src/main/java/com/conveyal/analysis/controllers/NetworkTileController.java

```diff
@@ -0,0 +1,158 @@
+package com.conveyal.analysis.controllers;
+
+import com.conveyal.analysis.util.JsonUtil;
+import com.conveyal.analysis.util.MapTile;
+import com.conveyal.r5.common.GeometryUtils;
+import com.conveyal.r5.streets.EdgeStore;
+import com.conveyal.r5.transit.TransportNetwork;
+import com.conveyal.r5.transit.TransportNetworkCache;
+import com.google.common.collect.ImmutableMap;
+import com.wdtinc.mapbox_vector_tile.adapt.jts.MvtEncoder;
+import com.wdtinc.mapbox_vector_tile.adapt.jts.UserDataKeyValueMapConverter;
+import com.wdtinc.mapbox_vector_tile.adapt.jts.model.JtsLayer;
+import com.wdtinc.mapbox_vector_tile.adapt.jts.model.JtsMvt;
+import com.wdtinc.mapbox_vector_tile.build.MvtLayerParams;
+import gnu.trove.set.TIntSet;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.CoordinateSequence;
+import org.locationtech.jts.geom.Envelope;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.LineString;
+import org.locationtech.jts.index.strtree.STRtree;
+import org.locationtech.jts.simplify.DouglasPeuckerSimplifier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.conveyal.analysis.controllers.GtfsVectorTileMaker.clipScaleAndSimplify;
+import static com.conveyal.analysis.util.HttpStatus.OK_200;
+import static com.conveyal.analysis.util.HttpUtils.CACHE_CONTROL_IMMUTABLE;
+import static com.conveyal.r5.common.GeometryUtils.floatingWgsEnvelopeToFixed;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Defines HTTP API endpoints to return Mapbox vector tiles of the street network. This is not simply the underlying
+ * OSM data, but the street edges created from that OSM data. This allows the user to verify how the OSM tags have
+ * been interpreted as mode permissions, speeds, etc. and eventually should allow visualizing the effets of scenarios
+ * on the street nework, includng scenarios that change the characteristics of street edges or create new streets.
+ * See GtfsVectorTileMaker for more information on the vector tile spec and tile numbers.
+ */
+public class NetworkTileController implements HttpController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(NetworkTileController.class);
+
+    /**
+     * Complex street geometries will be simplified, but the geometry will not deviate from the original by more
+     * than this many tile units. These are minuscule at 1/4096 of the tile width or height.
+     */
+    private static final int LINE_SIMPLIFY_TOLERANCE = 5;
```

Comment:
Should `clipScaleAndSimplify` and common static values like `LINE_SIMPLIFY_TOLERANCE`, `TILE_EXTENT`, and `TILE_SIZE` be factored out into a `TileUtils` file?

Reply:
Summarizing discussion from today's call: they're not quite static utility methods, as the goal is to reuse the envelope and buffered envelope of a particular tile as fields on an object. So we want to factor them out, but to a concrete class in a vector tile-specific package.